In [402]:
import gzip
import pandas as pd

In [403]:
# nm=gzip.open('IMDB/name.basics.tsv.gz','rb')
# df_names = pd.read_csv(nm,sep='\t', low_memory=False)
# df_names.head(10)


# # Index nconst	primaryName	birthYear	deathYear	primaryProfession	knownForTitles
# # 0	nm0000001	Fred Astaire	1899	1987	soundtrack,actor,miscellaneous	tt0050419,tt0031983,tt0072308,tt0053137
# # 1	nm0000002	Lauren Bacall	1924	2014	actress,soundtrack	tt0117057,tt0071877,tt0038355,tt0037382
# # 2	nm0000003	Brigitte Bardot	1934	\N	actress,soundtrack,music_department	tt0057345,tt0049189,tt0056404,tt0054452
# # 3	nm0000004	John Belushi	1949	1982	actor,soundtrack,writer	tt0072562,tt0077975,tt0078723,tt0080455
# # 4	nm0000005	Ingmar Bergman	1918	2007	writer,director,actor	tt0083922,tt0069467,tt0060827,tt0050986
# # 5	nm0000006	Ingrid Bergman	1915	1982	actress,soundtrack,producer	tt0077711,tt0038109,tt0036855,tt0034583
# # 6	nm0000007	Humphrey Bogart	1899	1957	actor,soundtrack,producer	tt0037382,tt0043265,tt0042593,tt0034583
# # 7	nm0000008	Marlon Brando	1924	2004	actor,soundtrack,director	tt0078788,tt0068646,tt0070849,tt0047296
# # 8	nm0000009	Richard Burton	1925	1984	actor,soundtrack,producer	tt0057877,tt0087803,tt0061184,tt0059749
# # 9	nm0000010	James Cagney	1899	1986	actor,soundtrack,director	tt0031867,tt0029870,tt0042041,tt0035575

In [404]:
# aka=gzip.open('IMDB/title.akas.tsv.gz','rb')
# df_akas = pd.read_csv(aka,sep='\t', low_memory=False)
# df_akas.head(10)


# # index titleId	ordering	title	region	language	types	attributes	isOriginalTitle
# # 0	tt0000001	1	Карменсіта	UA	\N	imdbDisplay	\N	0
# # 1	tt0000001	2	Carmencita	DE	\N	\N	literal title	0
# # 2	tt0000001	3	Carmencita - spanyol tánc	HU	\N	imdbDisplay	\N	0
# # 3	tt0000001	4	Καρμενσίτα	GR	\N	imdbDisplay	\N	0
# # 4	tt0000001	5	Карменсита	RU	\N	imdbDisplay	\N	0
# # 5	tt0000001	6	Carmencita	US	\N	imdbDisplay	\N	0
# # 6	tt0000001	7	Carmencita	\N	\N	original	\N	1
# # 7	tt0000001	8	カルメンチータ	JP	ja	imdbDisplay	\N	0
# # 8	tt0000002	1	Le clown et ses chiens	\N	\N	original	\N	1
# # 9	tt0000002	2	Le clown et ses chiens	FR	\N	imdbDisplay	\N	0

In [405]:
# cr=gzip.open('IMDB/title.crew.tsv.gz','rb')
# df_crews = pd.read_csv(cr,sep='\t', low_memory=False)
# df_crews.tail(10)


# # index tconst	directors	writers
# # 8784762	tt9916838	nm5519375,nm5519454	nm6182221,nm1628284,nm2921377
# # 8784763	tt9916840	nm0996406	nm1482639,nm2586970
# # 8784764	tt9916842	nm5519454,nm5519375	nm6182221,nm1628284,nm2921377
# # 8784765	tt9916844	nm5519454,nm5519375	nm6182221,nm1628284,nm2921377
# # 8784766	tt9916846	nm5519375,nm5519454	nm6182221,nm1628284,nm2921377
# # 8784767	tt9916848	nm5519454,nm5519375	nm6182221,nm1628284,nm2921377
# # 8784768	tt9916850	nm5519454,nm5519375	nm6182221,nm1628284,nm2921377
# # 8784769	tt9916852	nm5519454,nm5519375	nm6182221,nm1628284,nm2921377
# # 8784770	tt9916856	nm10538645	nm6951431
# # 8784771	tt9916880	nm0996406	nm1482639,nm2586970

In [406]:
# since we are not considering anything other than full-length features, this may not be relevant
# ep=gzip.open('IMDB/title.episode.tsv.gz','rb')
# df_episodes = pd.read_csv(ep,sep='\t', low_memory=False)
# df_episodes.head()

# # Index tconst	parentTconst	seasonNumber	episodeNumber
# # 0	tt0020666	tt15180956	1	2
# # 1	tt0020829	tt15180956	1	1
# # 2	tt0021166	tt15180956	1	3
# # 3	tt0021612	tt15180956	2	2
# # 4	tt0021655	tt15180956	2	5

In [407]:
# pr=gzip.open('IMDB/title.principals.tsv.gz','rb')
# df_principals = pd.read_csv(pr,sep='\t', low_memory=False)
# df_principals.head()

# # Index tconst	ordering	nconst	category	job	characters
# # 0	tt0000001	1	nm1588970	self	\N	["Self"]
# # 1	tt0000001	2	nm0005690	director	\N	\N
# # 2	tt0000001	3	nm0374658	cinematographer	director of photography	\N
# # 3	tt0000002	1	nm0721526	director	\N	\N
# # 4	tt0000002	2	nm1335271	composer	\N	\N

In [408]:
# rt=gzip.open('IMDB/title.ratings.tsv.gz','rb')
# df_ratings = pd.read_csv(rt,sep='\t', low_memory=False)
# df_ratings.head()
# # Index	tconst	averageRating	numVotes
# # 0	tt0000001	5.7	1868
# # 1	tt0000002	5.9	247
# # 2	tt0000003	6.5	1640
# # 3	tt0000004	5.8	159
# # 4	tt0000005	6.2	2463

# Get the titles from IMDB, the tconst values, for each movie on each of our three lists.  D.com, D23, and Wikipedia

In [409]:
bs=gzip.open('IMDB/title.basics.tsv.gz','rb')
df_basics = pd.read_csv(bs,sep='\t', low_memory=False)

In [465]:
df_basics.shape

(8784772, 9)

In [410]:
df_basics.head(20)

tconst titleType                                 primaryTitle  \
0   tt0000001     short                                   Carmencita   
1   tt0000002     short                       Le clown et ses chiens   
2   tt0000003     short                               Pauvre Pierrot   
3   tt0000004     short                                  Un bon bock   
4   tt0000005     short                             Blacksmith Scene   
5   tt0000006     short                            Chinese Opium Den   
6   tt0000007     short  Corbett and Courtney Before the Kinetograph   
7   tt0000008     short       Edison Kinetoscopic Record of a Sneeze   
8   tt0000009     short                                   Miss Jerry   
9   tt0000010     short                          Leaving the Factory   
10  tt0000011     short                      Akrobatisches Potpourri   
11  tt0000012     short                       The Arrival of a Train   
12  tt0000013     short  The Photographical Congress Arrives in Lyon   
13  tt0000014     short                          The Waterer Watered   
14  tt0000015     short                          Autour d'une cabine   
15  tt0000016     short                        Boat Leaving the Port   
16  tt0000017     short                     Italienischer Bauerntanz   
17  tt0000018     short                         Das boxende Känguruh   
18  tt0000019     short                             The Clown Barber   
19  tt0000020     short                               The Derby 1895   

                                        originalTitle isAdult startYear  \
0                                          Carmencita       0      1894   
1                              Le clown et ses chiens       0      1892   
2                                      Pauvre Pierrot       0      1892   
3                                         Un bon bock       0      1892   
4                                    Blacksmith Scene       0      1893   
5                                   Chinese Opium Den       0      1894   
6         Corbett and Courtney Before the Kinetograph       0      1894   
7              Edison Kinetoscopic Record of a Sneeze       0      1894   
8                                          Miss Jerry       0      1894   
9                 La sortie de l'usine Lumière à Lyon       0      1895   
10                            Akrobatisches Potpourri       0      1895   
11                   L'arrivée d'un train à La Ciotat       0      1896   
12  Le débarquement du congrès de photographie à Lyon       0      1895   
13                                  L'arroseur arrosé       0      1895   
14                                Autour d'une cabine       0      1894   
15                             Barque sortant du port       0      1895   
16                           Italienischer Bauerntanz       0      1895   
17                               Das boxende Känguruh       0      1895   
18                                   The Clown Barber       0      1898   
19                                     The Derby 1895       0      1895   

   endYear runtimeMinutes                    genres  
0       \N              1         Documentary,Short  
1       \N              5           Animation,Short  
2       \N              4  Animation,Comedy,Romance  
3       \N             12           Animation,Short  
4       \N              1              Comedy,Short  
5       \N              1                     Short  
6       \N              1               Short,Sport  
7       \N              1         Documentary,Short  
8       \N             40             Romance,Short  
9       \N              1         Documentary,Short  
10      \N              1         Documentary,Short  
11      \N              1         Documentary,Short  
12      \N              1         Documentary,Short  
13      \N              1              Comedy,Short  
14      \N              2           Animation,Short  
15      \N              1         Documentary,Short  


In [411]:
FL_Disney = pd.read_csv('DD_Films_List_Disney_Com/Film_list_Disney_com.csv')
FL_Disney.rename(     columns=({ 'Movie Title': 'title'}),     inplace=True )

print(FL_Disney.shape)
FL_Disney.head()

(657, 1)


title
0                               101 Dalmatians
1                        101 Dalmatians (1996)
2  101 Dalmatians II: Patch's London Adventure
3                               102 Dalmatians
4                 20,000 Leagues Under the Sea

First merge of the Disney list and the IMDB data.

In [412]:
FL_Disney_IMDB = pd.merge(FL_Disney ,                 # left df
                          df_basics,                  # right df
                          how="left",                 # left join
                          left_on='title',            # left column
                          right_on='primaryTitle',    # right column
                          indicator = True,           # indicates source of each row
                          #validate = "one_to_many"    # alerts us of the relationship from left to right, incase there are dups
        )

In [413]:
FL_Disney_IMDB.head()

title     tconst  titleType  \
0                               101 Dalmatians  tt0115433      movie   
1                               101 Dalmatians  tt1185295  tvEpisode   
2                               101 Dalmatians  tt5289602  tvEpisode   
3                        101 Dalmatians (1996)  tt6358998  tvEpisode   
4  101 Dalmatians II: Patch's London Adventure  tt6329102  tvEpisode   

                                  primaryTitle  \
0                               101 Dalmatians   
1                               101 Dalmatians   
2                               101 Dalmatians   
3                        101 Dalmatians (1996)   
4  101 Dalmatians II: Patch's London Adventure   

                                 originalTitle isAdult startYear endYear  \
0                               101 Dalmatians       0      1996      \N   
1                               101 Dalmatians       0      2008      \N   
2                               101 Dalmatians       0      2011      \N   
3                        101 Dalmatians (1996)       0      2012      \N   
4  101 Dalmatians II: Patch's London Adventure       0      2016      \N   

  runtimeMinutes                  genres _merge  
0            103  Adventure,Comedy,Crime   both  
1             \N                  Family   both  
2              3       Documentary,Short   both  
3             \N               Talk-Show   both  
4             \N       Documentary,Short   both

In [414]:
print(FL_Disney_IMDB.shape)
FL_Disney_IMDB['_merge'].value_counts()

(6143, 11)


both          6018
left_only      125
right_only       0
Name: _merge, dtype: int64

In [415]:
FL_Disney_IMDB['titleType'].value_counts()

tvEpisode       3675
movie            904
short            668
tvMovie          244
video            218
tvSeries         160
videoGame         83
tvMiniSeries      56
tvShort            6
tvSpecial          4
Name: titleType, dtype: int64

We started with 657 rows, and ended with 6143, so there are a lot of duplicates that we'll need to clean up.  125 title from the original Disney list were not found in the IMDM list. To clean this up, we'll pull at the 125 from the Disney list, and explore those seperately.  We'll pull out any single matches from the list, and explore those under the hypothesis that they are correct but will look to disprove that hypothesis.  Finally, we'll look at the duplicated values from the list and attempt to ID which titles we really need to keep.  The goal is to get back down the original 657 row with complete IMDB data.  Since we only want to consider feature releases, not TV or Video, then we will likely end up with fewwe than the original 657 AFTER we've done some filtering.

First, lets look at the 125 that were not found in IMDB at all.

Creating a temp df.  We will update the tconst and the primaryTitle on the temp df as we fix things.  Then we'll strip out  the index, title, tconst, and primaryTitle

In [417]:
FL_Disney_not_IMDB = FL_Disney_IMDB[FL_Disney_IMDB['_merge'] == "left_only"]

In [418]:
FL_Disney_not_IMDB.head()

title tconst titleType primaryTitle  \
132     A Kid In King Arthur's Court    NaN       NaN          NaN   
133              A Knight In Camelot    NaN       NaN          NaN   
134              A Mom For Christmas    NaN       NaN          NaN   
138  A Very Playhouse Disney Holiday    NaN       NaN          NaN   
306     Aladdin: The Return of Jafar    NaN       NaN          NaN   

    originalTitle isAdult startYear endYear runtimeMinutes genres     _merge  
132           NaN     NaN       NaN     NaN            NaN    NaN  left_only  
133           NaN     NaN       NaN     NaN            NaN    NaN  left_only  
134           NaN     NaN       NaN     NaN            NaN    NaN  left_only  
138           NaN     NaN       NaN     NaN            NaN    NaN  left_only  
306           NaN     NaN       NaN     NaN            NaN    NaN  left_only

Spot checking the first 5:
 - A Kid In King Arthur's Court: Appears on IMDB.com, 1995, PG, 1h 29m #tt0113538 <mark>Will search the IMDB df</mark>
 - A Knight In Camelot: Appears on IMDB.com, TV Film, considered S2: E6 of "The Wonderful World of Disney", 1998, NR, 1h 30m #tt0176892 <mark>We want to consider feature films. Not TV movies.</mark>
 - A Mom For Christmas: Appears on IMDB.com, 1990, TV-G, 1h 36m #tt0100173 <mark>We want to consider feature films. Not TV movies.</mark>
 - A Very Playhouse Disney Holiday, NOT on IMDB.com, looks like a DVD release, 2005, NR, 56m <mark>We want to consider feature films. Not video releases.</mark>
 - Aladdin: The Return of Jafar: Appears on IMDB.com as "Aladdin 2: The Return of Jafar" but video, not a feature film, 1994, G, 1h 9m #tt0107952 <mark>We want to consider feature films. Not video releases.</mark>

In [419]:
spot_check = ['tt0113538', 'tt0176892', 'tt0100173', 'tt0107952']
df_basics[df_basics['tconst'].isin(spot_check)]

tconst  titleType                    primaryTitle  \
97950   tt0100173    tvMovie             A Mom for Christmas   
105529  tt0107952      video  Aladdin 2: The Return of Jafar   
110982  tt0113538      movie    A Kid in King Arthur's Court   
170769  tt0176892  tvEpisode             A Knight in Camelot   

                       originalTitle isAdult startYear endYear runtimeMinutes  \
97950            A Mom for Christmas       0      1990      \N             96   
105529           The Return of Jafar       0      1994      \N             69   
110982  A Kid in King Arthur's Court       0      1995      \N             89   
170769           A Knight in Camelot       0      1998      \N             90   

                            genres  
97950        Comedy,Family,Fantasy  
105529  Adventure,Animation,Comedy  
110982     Adventure,Comedy,Family  
170769      Adventure,Comedy,Drama

The match appears to be case sensitive, notice the "i" vs "I" in "in/In".

In [420]:
"A Kid In King Arthur's Court"=="A Kid in King Arthur's Court"

False

notice the "i" vs "I" in "in/In"

In [421]:
"A Kid In King Arthur's Court"=="A Kid in King Arthur's Court"

False

notice the "f" vs "F" in "for/For".

In [422]:
"A Mom For Christmas"=="A Mom for Christmas"

False

Some of the Disney list appears in true title format, with words like "In" and "For" being capitalized. We will apply .title() to both the Disney and IMDB lists, then attempt the join again.

In [423]:
FL_Disney['title'] = FL_Disney['title'].str.title()

In [424]:
df_basics['primaryTitle'] = df_basics['primaryTitle'].str.title()

Attempting the merge again.

In [425]:
FL_Disney_IMDB_temp = pd.merge(FL_Disney ,                 # left df
                          df_basics,                  # right df
                          how="left",                 # left join
                          left_on='title',            # left column
                          right_on='primaryTitle',    # right column
                          indicator = True,           # indicates source of each row
                          #validate = "one_to_many"    # alerts us of the relationship from left to right, incase there are dups
        )

In [426]:
FL_Disney_IMDB_temp.head()

title     tconst  titleType  \
0                               101 Dalmatians  tt0115433      movie   
1                               101 Dalmatians  tt1185295  tvEpisode   
2                               101 Dalmatians  tt5289602  tvEpisode   
3                        101 Dalmatians (1996)  tt6358998  tvEpisode   
4  101 Dalmatians Ii: Patch'S London Adventure  tt6329102  tvEpisode   

                                  primaryTitle  \
0                               101 Dalmatians   
1                               101 Dalmatians   
2                               101 Dalmatians   
3                        101 Dalmatians (1996)   
4  101 Dalmatians Ii: Patch'S London Adventure   

                                 originalTitle isAdult startYear endYear  \
0                               101 Dalmatians       0      1996      \N   
1                               101 Dalmatians       0      2008      \N   
2                               101 Dalmatians       0      2011      \N   
3                        101 Dalmatians (1996)       0      2012      \N   
4  101 Dalmatians II: Patch's London Adventure       0      2016      \N   

  runtimeMinutes                  genres _merge  
0            103  Adventure,Comedy,Crime   both  
1             \N                  Family   both  
2              3       Documentary,Short   both  
3             \N               Talk-Show   both  
4             \N       Documentary,Short   both

In [427]:
print(FL_Disney_IMDB_temp.shape)
FL_Disney_IMDB_temp['_merge'].value_counts()

(6293, 11)


both          6209
left_only       84
right_only       0
Name: _merge, dtype: int64

In [428]:
FL_Disney_not_IMDB = FL_Disney_IMDB_temp[FL_Disney_IMDB_temp['tconst'].isnull()]

In [429]:
FL_Disney_not_IMDB.head(50)

title tconst titleType  \
141                     A Very Playhouse Disney Holiday    NaN       NaN   
313                        Aladdin: The Return Of Jafar    NaN       NaN   
489                            America'S Heart And Soul    NaN       NaN   
559                                        Annie (1999)    NaN       NaN   
628                                   Atta Girl, Kelly!    NaN       NaN   
1427                        Charlie The Lonesome Cougar    NaN       NaN   
1482  Chronicles Of Narnia: The Lion, The Witch And ...    NaN       NaN   
1758  Disney Animation Collection Volume 1: Mickey A...    NaN       NaN   
1759  Disney Animation Collection Volume 2: Three Li...    NaN       NaN   
1760  Disney Animation Collection Volume 3: The Prin...    NaN       NaN   
1761  Disney Animation Collection Volume 4: The Tort...    NaN       NaN   
1762  Disney Animation Collection Volume 5: Wind In ...    NaN       NaN   
1763  Disney Animation Collection Volume 6: The Relu...    NaN       NaN   
1764  Disney Animation Collection Volume 7: Mickey'S...    NaN       NaN   
1766                              Disney Junior Holiday    NaN       NaN   
1767  Disney Learning Adventures: Mickey And The Bea...    NaN       NaN   
1768  Disney Learning Adventures: Mickey'S Around Th...    NaN       NaN   
1770                  Disney Princess Party: Volume One    NaN       NaN   
1772  Disney Princess Sing Along Songs Volume One: O...    NaN       NaN   
1773  Disney Princess Sing Along Songs Volume Three:...    NaN       NaN   
1774  Disney Princess Sing Along Songs Volume Two: E...    NaN       NaN   
1778            Disney Sing Along Songs: Happy Haunting    NaN       NaN   
1780  Disney Sing-Along-Songs: Beach Party At Walt D...    NaN       NaN   
1785                          Disney'S American Legends    NaN       NaN   
1787                            Doc Mcstuffins: Pet Vet    NaN       NaN   
1790                      Dr. Seuss' Horton Hears A Who    NaN       NaN   
1791   Ducktales: The Movie - Treasure Of The Lost Lamp    NaN       NaN   
1871    Elfego Baca And The Swamp Fox, Legendary Heroes    NaN       NaN   
2141                               Freaky Friday (1976)    NaN       NaN   
2210                        Frozen (Sing-Along Edition)    NaN       NaN   
2304  Hannah Montana & Miley Cyrus: Best Of Both Wor...    NaN       NaN   
2305                   Hannah Montana: Pop Star Profile    NaN       NaN   
2314                 Hans Brinker, Or The Silver Skates    NaN       NaN   
2316             Have Yourself A Goofy Little Christmas    NaN       NaN   
2363                               Herbie: Fully Loaded    NaN       NaN   
2429                         High School Musical: China    NaN       NaN   
2430                   High School Musical: The Concert    NaN       NaN   
2431                     Hilary Duff: The Girl Can Rock    NaN       NaN   
2937             Jonas Brothers: The Concert Experience    NaN       NaN   
3283                     Marvel Studios' Doctor Strange    NaN       NaN   
3284                           Marvel Studios' Eternals    NaN       NaN   
3285                   Marvel'S Avengers: Age Of Ultron    NaN       NaN   
3286           Marvel'S Captain America: Winter Soldier    NaN       NaN   
3290                                Marvel'S Iron Man 3    NaN       NaN   
3339  Mickey Mouse Clubhouse And The Quest For The C...    NaN       NaN   
3340          Mickey Mouse Clubhouse: Choo-Choo Express    NaN       NaN   
3342               Mickey Mouse Clubhouse: Minnie-Rella    NaN       NaN   
3343                 Mickey Mouse Clubhouse: Road Rally    NaN       NaN   
3344            Mickey Mouse Clubhouse: Space Adventure    NaN       NaN   
3345                    Mickey Mouse In Black And White    NaN       NaN   

     primaryTitle originalTitle isAdult startYear endYear runtimeMinutes  \
141           NaN           NaN     NaN       NaN     NaN            NaN   
313           NaN    

Since we kept the index values from the Disney list, we could use the index to update all of the "titleType"s for the TV movies or video movies <br>
FL_Disney_IMDB.at[141,'titleType'] = 'tvMovie' <BR>
However, for the sake of take, we won't bother updating any titles that we don't intend to use in our analysis.  We will scan the new list of 84 titles for feature releases and update them in the temp df. 

In [430]:
FL_Disney_IMDB_temp.at[489,'primaryTitle'] = "America's Heart And Soul"    # America'S Heart And Soul tt0381006
FL_Disney_IMDB_temp.at[489,'tconst'] = "tt0381006"    # America'S Heart And Soul tt0381006

FL_Disney_IMDB_temp.at[559,'primaryTitle'] = "Annie"               # Annie (1999) tt0207972
FL_Disney_IMDB_temp.at[559,'tconst'] = "tt0207972"               # Annie (1999) tt0207972

# Atta Girl, Kelly!   tt0560969

FL_Disney_IMDB_temp.at[1427,'primaryTitle'] = "Charlie, the Lonesome Cougar"  #   Charlie The Lonesome Cougar tt0062793
FL_Disney_IMDB_temp.at[1427,'tconst'] = "tt0062793"  #   Charlie The Lonesome Cougar tt0062793

FL_Disney_IMDB_temp.at[1482,'primaryTitle'] = "The Chronicles of Narnia: The Lion, the Witch and the Wardrobe" # Chronicles Of Narnia: The Lion, The Witch And  tt0363771
FL_Disney_IMDB_temp.at[1482,'tconst'] = "tt0363771" # Chronicles Of Narnia: The Lion, The Witch And  tt0363771

FL_Disney_IMDB_temp.at[1785,'primaryTitle'] = "American Legends"   # Disney'S American Legends tt0372866
FL_Disney_IMDB_temp.at[1785,'tconst'] = "tt0372866"   # Disney'S American Legends tt0372866

FL_Disney_IMDB_temp.at[1790,'primaryTitle'] = "Horton Hears a Who!" # Dr. Seuss' Horton Hears A Who tt0451079
FL_Disney_IMDB_temp.at[1790,'tconst'] = "tt0451079" # Dr. Seuss' Horton Hears A Who tt0451079

FL_Disney_IMDB_temp.at[1791,'primaryTitle'] = "DuckTales the Movie: Treasure of the Lost Lamp" # 1791	Ducktales: The Movie - Treasure Of The Lost Lamp	tt0099472
FL_Disney_IMDB_temp.at[1791,'tconst'] = "tt0099472" # 1791	Ducktales: The Movie - Treasure Of The Lost Lamp	tt0099472

FL_Disney_IMDB_temp.at[2141,'primaryTitle'] = "Freaky Friday" # 2141	Freaky Friday (1976)	tt0076054
FL_Disney_IMDB_temp.at[2141,'tconst'] = "tt0076054" # 2141	Freaky Friday (1976)	tt0076054

FL_Disney_IMDB_temp.at[2304,'primaryTitle'] = "Hannah Montana and Miley Cyrus: Best of Both Worlds Concert" # 2304	Hannah Montana & Miley Cyrus: Best Of Both Wor...	tt1127884
FL_Disney_IMDB_temp.at[2304,'tconst'] = "tt1127884" # 2304	Hannah Montana & Miley Cyrus: Best Of Both Wor... tt1127884

FL_Disney_IMDB_temp.at[2363,'primaryTitle'] = "Herbie Fully Loaded" # 2363	Herbie: Fully Loaded	tt0400497
FL_Disney_IMDB_temp.at[2363,'tconst'] = "tt0400497" # 2363	Herbie: Fully Loaded  tt0400497

FL_Disney_IMDB_temp.at[2429,'primaryTitle'] = "Disney High School Musical: China" # 2429	High School Musical: China	tt1556143
FL_Disney_IMDB_temp.at[2429,'tconst'] = "tt1556143" # 2429	High School Musical: China  tt1556143

FL_Disney_IMDB_temp.at[2937,'primaryTitle'] = "Jonas Brothers: The 3D Concert Experience" # 2937	Jonas Brothers: The Concert Experience	tt1229827
FL_Disney_IMDB_temp.at[2937,'tconst'] = "tt1229827" # 2937	Jonas Brothers: The Concert Experience  tt1229827

FL_Disney_IMDB_temp.at[3283,'primaryTitle'] = "Doctor Strange" # 3283	Marvel Studios' Doctor Strange		tt1211837
FL_Disney_IMDB_temp.at[3283,'tconst'] = "tt1211837" # 3283	Marvel Studios' Doctor Strange	  tt1211837

FL_Disney_IMDB_temp.at[3285,'primaryTitle'] = "Avengers: Age of Ultron" # 3285	Marvel'S Avengers: Age Of Ultron		tt2395427
FL_Disney_IMDB_temp.at[3285,'tconst'] = "tt2395427" # 3285	Marvel'S Avengers: Age Of Ultron	  tt2395427

FL_Disney_IMDB_temp.at[3286,'primaryTitle'] = "Captain America: The Winter Soldier" # 33286	Marvel'S Captain America: Winter Soldier		tt1843866
FL_Disney_IMDB_temp.at[3286,'tconst'] = "tt1843866" # 3286	Marvel'S Captain America: Winter Soldier	  tt1843866

FL_Disney_IMDB_temp.at[3290,'primaryTitle'] = "Iron Man 3" # 3290	Marvel'S Iron Man 3 	tt1300854
FL_Disney_IMDB_temp.at[3290,'tconst'] = "tt1300854" # 3290	Marvel'S Iron Man 3	  tt1300854

FL_Disney_IMDB_temp.at[4045,'primaryTitle'] = "Ralph Breaks the Internet" # 4045	Ralph Breaks The Internet: Wreck-It Ralph 2 	tt5848272
FL_Disney_IMDB_temp.at[4045,'tconst'] = "tt5848272" # 4045	Ralph Breaks The Internet: Wreck-It Ralph 2	  tt5848272

FL_Disney_IMDB_temp.at[4091,'primaryTitle'] = "Peter Pan 2: Return to Never Land" # 4091	Return To Never Land	tt0280030
FL_Disney_IMDB_temp.at[4091,'tconst'] = "tt0280030" # 4091	Return To Never Land		  tt0280030

FL_Disney_IMDB_temp.at[4465,'primaryTitle'] = "Sleeping Beauty" # 4465	Sleeping Beauty (1959)	tt0053285tt0053285
FL_Disney_IMDB_temp.at[4465,'tconst'] = "tt0053285" # 4465	Sleeping Beauty (1959)	tt0053285

FL_Disney_IMDB_temp.at[4761,'primaryTitle'] = "Tall Tale" # 4761	Tall Tale: The Unbelievable Adventure	tt0111359
FL_Disney_IMDB_temp.at[4761,'tconst'] = "tt0111359" # 4761	Tall Tale: The Unbelievable Adventure	tt0111359

FL_Disney_IMDB_temp.at[5008,'primaryTitle'] = "The Boys" # 5008	The Boys: The Sherman Brothers' Story tt1015971
FL_Disney_IMDB_temp.at[5008,'tconst'] = "tt1015971" # 5008	The Boys: The Sherman Brothers' Story tt1015971

FL_Disney_IMDB_temp.at[5891,'primaryTitle'] = "Toby Tyler or Ten Weeks with a Circus" # 5891	Toby Tyler	   tt0054390
FL_Disney_IMDB_temp.at[5891,'tconst'] = "tt0054390" # 5891	Toby Tyler	   tt0054390

FL_Disney_IMDB_temp.at[6255,'primaryTitle'] = "Disneynature: Wings of Life" # 6255	Wings Of Life	  tt1222816
FL_Disney_IMDB_temp.at[6255,'tconst'] = "tt1222816" # 6255	Wings Of Life	tt1222816

FL_Disney_IMDB_temp.at[6256,'primaryTitle'] = "Winnie the Pooh" # 6256	Winnie The Pooh (2011)	  tt1449283
FL_Disney_IMDB_temp.at[6256,'tconst'] = "tt1449283" # 6256	Winnie The Pooh (2011)	 tt1449283
   
      
    


In [431]:
FL_Disney_IMDB_temp.head()

title     tconst  titleType  \
0                               101 Dalmatians  tt0115433      movie   
1                               101 Dalmatians  tt1185295  tvEpisode   
2                               101 Dalmatians  tt5289602  tvEpisode   
3                        101 Dalmatians (1996)  tt6358998  tvEpisode   
4  101 Dalmatians Ii: Patch'S London Adventure  tt6329102  tvEpisode   

                                  primaryTitle  \
0                               101 Dalmatians   
1                               101 Dalmatians   
2                               101 Dalmatians   
3                        101 Dalmatians (1996)   
4  101 Dalmatians Ii: Patch'S London Adventure   

                                 originalTitle isAdult startYear endYear  \
0                               101 Dalmatians       0      1996      \N   
1                               101 Dalmatians       0      2008      \N   
2                               101 Dalmatians       0      2011      \N   
3                        101 Dalmatians (1996)       0      2012      \N   
4  101 Dalmatians II: Patch's London Adventure       0      2016      \N   

  runtimeMinutes                  genres _merge  
0            103  Adventure,Comedy,Crime   both  
1             \N                  Family   both  
2              3       Documentary,Short   both  
3             \N               Talk-Show   both  
4             \N       Documentary,Short   both

We can see the 24 titles that we've updated

In [432]:
FL_Disney_IMDB_temp.isnull().sum()

title              0
tconst            60
titleType         84
primaryTitle      60
originalTitle     84
isAdult           84
startYear         84
endYear           84
runtimeMinutes    84
genres            84
_merge             0
dtype: int64

In [433]:
FL_Disney_IMDB_temp.drop(columns=['titleType','originalTitle','primaryTitle','isAdult','startYear','endYear','runtimeMinutes','genres','_merge'], inplace=True)

In [441]:
FL_Disney_IMDB_temp.head()

title     tconst
0                               101 Dalmatians  tt0115433
1                               101 Dalmatians  tt1185295
2                               101 Dalmatians  tt5289602
3                        101 Dalmatians (1996)  tt6358998
4  101 Dalmatians Ii: Patch'S London Adventure  tt6329102

The index is the original index from our Disney List, 'tconst' is the unique identifier from IMDB.  Now we need to go back to the IMDB data, pull in the meta data, and deal with the duplicates.  

In [442]:
FL_Disney_IMDB_temp_2 = pd.merge(FL_Disney_IMDB_temp ,                 # left df
                          df_basics,                  # right df
                          how="left",                 # left join
                          left_on='tconst',            # left column
                          right_on='tconst',    # right column
                          indicator = True,           # indicates source of each row
                          #validate = "one_to_many"    # alerts us of the relationship from left to right, incase there are dups
        )

In [443]:
FL_Disney_IMDB_temp_2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6293 entries, 0 to 6292
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   title           6293 non-null   object  
 1   tconst          6233 non-null   object  
 2   titleType       6233 non-null   object  
 3   primaryTitle    6233 non-null   object  
 4   originalTitle   6233 non-null   object  
 5   isAdult         6233 non-null   object  
 6   startYear       6233 non-null   object  
 7   endYear         6233 non-null   object  
 8   runtimeMinutes  6233 non-null   object  
 9   genres          6233 non-null   object  
 10  _merge          6293 non-null   category
dtypes: category(1), object(10)
memory usage: 547.1+ KB


We started with 657, so there is a lot of duplcation in here. 

In [444]:
FL_Disney_IMDB_temp_2.titleType.value_counts()

tvEpisode       3789
movie            963
short            688
tvMovie          254
video            225
tvSeries         165
videoGame         83
tvMiniSeries      56
tvShort            6
tvSpecial          4
Name: titleType, dtype: int64

For this project, we want to focus on movies.  Anything tv, video, or short can come out.

In [445]:
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2[FL_Disney_IMDB_temp_2.titleType == 'movie']

In [446]:
FL_Disney_IMDB_temp_2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 963 entries, 0 to 6281
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   title           963 non-null    object  
 1   tconst          963 non-null    object  
 2   titleType       963 non-null    object  
 3   primaryTitle    963 non-null    object  
 4   originalTitle   963 non-null    object  
 5   isAdult         963 non-null    object  
 6   startYear       963 non-null    object  
 7   endYear         963 non-null    object  
 8   runtimeMinutes  963 non-null    object  
 9   genres          963 non-null    object  
 10  _merge          963 non-null    category
dtypes: category(1), object(10)
memory usage: 83.8+ KB


In [447]:
FL_Disney_IMDB_temp_2._merge.value_counts()

both          963
left_only       0
right_only      0
Name: _merge, dtype: int64

In [448]:
FL_Disney_IMDB_temp_2.drop_duplicates(subset=['tconst'], inplace=True)

In [449]:
FL_Disney_IMDB_temp_2._merge.value_counts()

both          921
left_only       0
right_only      0
Name: _merge, dtype: int64

In [450]:
FL_Disney_IMDB_temp_2.head(20)

title      tconst titleType  \
0                 101 Dalmatians   tt0115433     movie   
5                 102 Dalmatians   tt0211181     movie   
9   20,000 Leagues Under The Sea   tt0006333     movie   
10  20,000 Leagues Under The Sea   tt0046672     movie   
13  20,000 Leagues Under The Sea   tt0397230     movie   
14  20,000 Leagues Under The Sea   tt0498755     movie   
16  20,000 Leagues Under The Sea  tt10915850     movie   
22  20,000 Leagues Under The Sea   tt9328210     movie   
23                  A Bug'S Life   tt0120623     movie   
36             A Christmas Carol   tt0029992     movie   
37             A Christmas Carol   tt0039562     movie   
38             A Christmas Carol   tt0044008     movie   
58             A Christmas Carol   tt0443734     movie   
70             A Christmas Carol  tt10415514     movie   
71             A Christmas Carol  tt10476682     movie   
73             A Christmas Carol   tt1067106     movie   
74             A Christmas Carol  tt10778144     movie   
84             A Christmas Carol  tt13569424     movie   
85             A Christmas Carol  tt13571948     movie   
92             A Christmas Carol  tt16048864     movie   

                    primaryTitle                             originalTitle  \
0                 101 Dalmatians                            101 Dalmatians   
5                 102 Dalmatians                            102 Dalmatians   
9   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
10  20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
13  20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
14  20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
16  20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
22  20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
23                  A Bug'S Life                              A Bug's Life   
36             A Christmas Carol                         A Christmas Carol   
37             A Christmas Carol                        Leyenda de Navidad   
38             A Christmas Carol                                   Scrooge   
58             A Christmas Carol  Natale a casa Deejay - A Christmas Carol   
70             A Christmas Carol                         A Christmas Carol   
71             A Christmas Carol                         A Christmas Carol   
73             A Christmas Carol                         A Christmas Carol   
74             A Christmas Carol                         A Christmas Carol   
84             A Christmas Carol                         A Christmas Carol   
85             A Christmas Carol                         A Christmas Carol   
92             A Christmas Carol                         A Christmas Carol   

   isAdult startYear endYear runtimeMinutes                      genres _merge  
0        0      1996      \N            103      Adventure,Comedy,Crime   both  
5        0      2000      \N            100     Adventure,Comedy,Family   both  
9        0      1916      \N            105     Action,Adventure,Sci-Fi   both  
10       0      1954      \N            127      Adventure,Drama,Family   both  
13       0      1973      \N             60            Animation,Family   both  
14       0      1985      \N             50  Action,Adventure,Animation   both  
16       0        \N      \N             \N                   Adventure   both  
22       0        \N      \N             \N                   Animation   both  
23       0      1998      \N             95  Adventure,Animation,Comedy   both  
36       0      1938      \N             69        Drama,Family,Fantasy   both  
37       0      1947      \N             80               Drama,Fantasy   both  
38       0      1951      \N             86        Drama,Family,Fantasy   both  
58       0      2004      \N             75              Comedy,Fantasy   both  
70       0        \N      \N       

Creating a list of the dupcate value.  Will have to seach IMDB to make sure we have the correct tconst going forward.  We could bump this up against one of the other source lists that include the year of release, but there's no guarantee that everything on this list will be on those, as we already know the counts are different, and there are some discrepancies in the month/year of release. Ultimately, this may prove faster than creating code and trying to account each variant.

In [451]:
dups_df = FL_Disney_IMDB_temp_2[FL_Disney_IMDB_temp_2['title'].isin(FL_Disney_IMDB_temp_2['title'].value_counts()[FL_Disney_IMDB_temp_2['title'].value_counts()>2].index)]
print(dups_df.shape)
dups_df.head(30)

(511, 11)


title      tconst titleType  \
9    20,000 Leagues Under The Sea   tt0006333     movie   
10   20,000 Leagues Under The Sea   tt0046672     movie   
13   20,000 Leagues Under The Sea   tt0397230     movie   
14   20,000 Leagues Under The Sea   tt0498755     movie   
16   20,000 Leagues Under The Sea  tt10915850     movie   
22   20,000 Leagues Under The Sea   tt9328210     movie   
36              A Christmas Carol   tt0029992     movie   
37              A Christmas Carol   tt0039562     movie   
38              A Christmas Carol   tt0044008     movie   
58              A Christmas Carol   tt0443734     movie   
70              A Christmas Carol  tt10415514     movie   
71              A Christmas Carol  tt10476682     movie   
73              A Christmas Carol   tt1067106     movie   
74              A Christmas Carol  tt10778144     movie   
84              A Christmas Carol  tt13569424     movie   
85              A Christmas Carol  tt13571948     movie   
92              A Christmas Carol  tt16048864     movie   
98              A Christmas Carol   tt2131447     movie   
100             A Christmas Carol   tt2294459     movie   
104             A Christmas Carol   tt4387314     movie   
106             A Christmas Carol   tt4776578     movie   
111             A Christmas Carol   tt5516774     movie   
118             A Christmas Carol   tt7820614     movie   
120             A Christmas Carol   tt8232648     movie   
121             A Christmas Carol   tt9136058     movie   
122             A Christmas Carol   tt9387380     movie   
123             A Christmas Carol   tt9416854     movie   
185                       Aladdin   tt0090601     movie   
186                       Aladdin   tt0103639     movie   
240                       Aladdin   tt6139732     movie   

                     primaryTitle                             originalTitle  \
9    20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
10   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
13   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
14   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
16   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
22   20,000 Leagues Under The Sea              20,000 Leagues Under the Sea   
36              A Christmas Carol                         A Christmas Carol   
37              A Christmas Carol                        Leyenda de Navidad   
38              A Christmas Carol                                   Scrooge   
58              A Christmas Carol  Natale a casa Deejay - A Christmas Carol   
70              A Christmas Carol                         A Christmas Carol   
71              A Christmas Carol                         A Christmas Carol   
73              A Christmas Carol                         A Christmas Carol   
74              A Christmas Carol                         A Christmas Carol   
84              A Christmas Carol                         A Christmas Carol   
85              A Christmas Carol                         A Christmas Carol   
92              A Christmas Carol                         A Christmas Carol   
98              A Christmas Carol                         A Christmas Carol   
100             A Christmas Carol                         A Christmas Carol   
104             A Christmas Carol                         A Christmas Carol   
106             A Christmas Carol                         A Christmas Carol   
111             A Christmas Carol                         A Christmas Carol   
118             A Christmas Carol                         A Christmas Carol   
120             A Christmas Carol                         A Christmas Carol   
121             A Christmas Carol                         A Christmas Carol   
122             A Christmas Carol                         A Christmas Carol   
123             A Christmas Carol                         A Chris

In [452]:
# 20,000 Leagues Under The Sea, keep tt0046672 
# A Christmas Carol, tt1067106

In [455]:
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "20,000 Leagues Under The Sea") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0046672")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "A Christmas Carol") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1067106")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Aladdin") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt6139732") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0103639"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Alice In Wonderland") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt0043274") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1014759"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Amy") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0082017")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Around The World In 80 Days") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0327437")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Bambi") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0034492")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Beauty And The Beast") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt11861230") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0101414"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Big Red") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0055793")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Black Widow") & (FL_Disney_IMDB_temp_2['tconst'] != "tt3480822")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Brave") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1217209")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Cheetah") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0097053")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Cinderella") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt0042332") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1661199"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Coco") & (FL_Disney_IMDB_temp_2['tconst'] != "tt2380307")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Dinosaur") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0130623")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Dumbo") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt0033563") & (FL_Disney_IMDB_temp_2['tconst'] != "tt3861390"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Earth") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0393597")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Epic") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0848537")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Emil And The Detectives") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0058056")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Enchanted") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0461770")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Endurance") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0120659")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Fantasia") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0032455")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Frozen") & (FL_Disney_IMDB_temp_2['tconst'] != "Frozen")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Gus") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0074599")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Heidi") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0107099")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Hercules") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0119282")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Home On The Range") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0299172")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Ice Age") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0268380")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "I'Ll Be Home For Christmas") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0155753")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "In Search Of The Castaways") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0056095")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Inside Out") & (FL_Disney_IMDB_temp_2['tconst'] != "tt2096673")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Into The Woods") & (FL_Disney_IMDB_temp_2['tconst'] != "tt2180411")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Invincible") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0445990")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Iron Will") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0110157")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Kidnapped") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0053994")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Luca") & (FL_Disney_IMDB_temp_2['tconst'] != "tt12801262")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Man Of The House") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0113755")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Midnight Madness") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0081159")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Miracle") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0349825	")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Moana") & (FL_Disney_IMDB_temp_2['tconst'] != "tt3521164")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Mulan") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt0120762") & (FL_Disney_IMDB_temp_2['tconst'] != "tt4566758"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Never A Dull Moment") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0063341")].index)
# *************************************************************************  This is a TV movie, and would have been removed at this point
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Oliver Twist") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0119825")].index)
# *************************************************************************
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Peter Pan") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0046183")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Pinocchio") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0032910")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Rascal") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0064875")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Rio") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1436562")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Robin Hood") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0070608")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Rocketman") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0120029")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Sky High") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0405325")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Snow White And The Seven Dwarfs") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0029583")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Something Wicked This Way Comes") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0086336")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Soul") & (FL_Disney_IMDB_temp_2['tconst'] != "tt2948372")].index)
# ************************************************************************* This is a TV movie, and would have been removed at this point
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Starstruck") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1579247")].index)
# *************************************************************************
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Swiss Family Robinson") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0054357")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Tangled") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0398286")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Teacher'S Pet") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0350194")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Black Hole") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0078869")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Duke") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0196516")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Hunchback Of Notre Dame") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0116583")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Kid") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0219854")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Lion King") & ( (FL_Disney_IMDB_temp_2['tconst'] != "tt0110357") & (FL_Disney_IMDB_temp_2['tconst'] != "tt6105098"))].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Little Mermaid") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0097757")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Lone Ranger") & (FL_Disney_IMDB_temp_2['tconst'] != "tt1210819")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Rookie") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0265662")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Sorcerer'S Apprentice") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0963966")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Three Musketeers") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0108333")].index)

FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "The Wild") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0405469")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Tonka") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0052300")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Treasure Island") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0043067")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Underdog") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0467110")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "Valiant") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0361089")].index)
FL_Disney_IMDB_temp_2 = FL_Disney_IMDB_temp_2.drop(FL_Disney_IMDB_temp_2[(FL_Disney_IMDB_temp_2['title'] == "White Fang") & (FL_Disney_IMDB_temp_2['tconst'] != "tt0103247")].index)

In [456]:
dups_df = FL_Disney_IMDB_temp_2[FL_Disney_IMDB_temp_2['title'].isin(FL_Disney_IMDB_temp_2['title'].value_counts()[FL_Disney_IMDB_temp_2['title'].value_counts()>2].index)]
print(dups_df.shape)
dups_df.head(30)

(0, 11)


Empty DataFrame
Columns: [title, tconst, titleType, primaryTitle, originalTitle, isAdult, startYear, endYear, runtimeMinutes, genres, _merge]
Index: []

In [461]:
FL_Disney_IMDB_temp_2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 483 entries, 0 to 6281
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   title           483 non-null    object  
 1   tconst          483 non-null    object  
 2   titleType       483 non-null    object  
 3   primaryTitle    483 non-null    object  
 4   originalTitle   483 non-null    object  
 5   isAdult         483 non-null    object  
 6   startYear       483 non-null    object  
 7   endYear         483 non-null    object  
 8   runtimeMinutes  483 non-null    object  
 9   genres          483 non-null    object  
 10  _merge          483 non-null    category
dtypes: category(1), object(10)
memory usage: 58.3+ KB


In [462]:
FL_Disney_IMDB_temp_2.head()

title     tconst titleType  \
0                 101 Dalmatians  tt0115433     movie   
5                 102 Dalmatians  tt0211181     movie   
10  20,000 Leagues Under The Sea  tt0046672     movie   
23                  A Bug'S Life  tt0120623     movie   
73             A Christmas Carol  tt1067106     movie   

                    primaryTitle                 originalTitle isAdult  \
0                 101 Dalmatians                101 Dalmatians       0   
5                 102 Dalmatians                102 Dalmatians       0   
10  20,000 Leagues Under The Sea  20,000 Leagues Under the Sea       0   
23                  A Bug'S Life                  A Bug's Life       0   
73             A Christmas Carol             A Christmas Carol       0   

   startYear endYear runtimeMinutes                      genres _merge  
0       1996      \N            103      Adventure,Comedy,Crime   both  
5       2000      \N            100     Adventure,Comedy,Family   both  
10      1954      \N            127      Adventure,Drama,Family   both  
23      1998      \N             95  Adventure,Animation,Comedy   both  
73      2009      \N             96  Adventure,Animation,Comedy   both

In [463]:
FL_Disney_IMDB_temp_2.to_csv('Combined_IMDB_Disney_483.csv')